[Detection Engine] Unskip some flaky tests, add better failure messages

rylnd opened the pull request:
## Summary

While the tests affected by this PR are varied, the changes contained here fall under one of two categories:

1. Unskipping flaky tests
2. Adding better test assertions (in order to produce more actionable failures later)

### Related Issues
* Closes #224699
* Closes #224780
* Closes #220943
* Closes #202940
* Closes #202945
* Closes #224780


### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios


<!--ONMERGE {"backportTargets":["8.19","9.1"]} ONMERGE-->

rylnd committed:
Replace KBN's expect for jest's

This generally has better matchers, and more verbose failure messages.
In this case, that is embodied in the use of `toHaveLength()` instead of
`expect(length).to.eql()`, which will display the array's contents upon
failure.

rylnd committed:
Unskip timestamp tests

I've been unable to reproduce these failures locally, and they fit the
pattern caused by our usage of es_archiver, and fixed in
https://github.com/elastic/kibana/pull/229457.

Closes #224780, closes #221659.

rylnd committed:
Unskip ESQL execution tests

A single test in this suite had failed (#224699) due to alerts not being
found. While this passes locally, this should give us a better error in
case it fails again.

Closes #224699.

rylnd committed:
Better assertion on flaky test

rylnd committed:
Use `toHaveLength` throughout flaky test suite

Regardless of why these are failing, this will result in better error
messages when they do.

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

rylnd committed:
Unskip flaky EQL rules

Similar to the other commits on this branch, we're unskipping tests
while improving the failure messages.

Closes #220943

elasticmachine commented:
Pinging @elastic/security-detection-engine (Team:Detection Engine)

rylnd committed:
Unskip and improve EQL suppression tests

* Unskips and closes #202940, closes #202945
* Replaces expect(length).toEqual() with expect().toHaveLength()

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/325760)
* Commit: 7aab65151d2a410e9c3856021bee9f1adcf5636f

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/325760","number":325760,"commit":"7aab65151d2a410e9c3856021bee9f1adcf5636f"}],"number":325760}
buildkite-pr-comment-->

rylnd committed:
Merge branch 'main' into rylnd/unskip_with_better_failures

rylnd committed:
Convert a few more assertions for verbosity

Particularly in the ML tests, which have been flaky recently.

rylnd committed:
Verbose failures in ESQL tests

These just started failing in
https://github.com/elastic/kibana/issues/230660, and we're again left
with a dead-end investigation due to the unhelpful error messages there.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/326670)
* Commit: 3de5c789072194a64772425966bf57a0116ebd17

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :yellow_heart: [Build #326085](https://buildkite.com/elastic/kibana-pull-request/builds/326085) was flaky 19c6bb4bf322ea6c5019f23664db93022abd9bcb
* :green_heart: [Build #325760](https://buildkite.com/elastic/kibana-pull-request/builds/325760) succeeded 7aab65151d2a410e9c3856021bee9f1adcf5636f

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326670","number":326670,"commit":"3de5c789072194a64772425966bf57a0116ebd17"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326085","number":326085,"commit":"19c6bb4bf322ea6c5019f23664db93022abd9bcb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/325760","number":325760,"commit":"7aab65151d2a410e9c3856021bee9f1adcf5636f"}],"number":326670}
buildkite-pr-comment-->

rylnd committed:
Merge branch 'main' into rylnd/unskip_with_better_failures

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/327203)
* Commit: 09db4a0ddf29dbe01b2572d6a8c18dc1c6849bcd

### Failed CI Steps
* [FTR Configs #42](https://buildkite.com/elastic/kibana-pull-request/builds/327203#019880fe-7493-4414-be65-2b1f71ba088b)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/327203#019880fe-7493-4414-be65-2b1f71ba088b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/327203/jobs/019880fe-7493-4414-be65-2b1f71ba088b/artifacts/01988133-015c-4ff9-a2f9-87f36a9a9211) FTR Configs #<span></span>42 / Interactive setup APIs - Manual configuration flow should be able to configure with valid authentication code

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #326670](https://buildkite.com/elastic/kibana-pull-request/builds/326670) succeeded 3de5c789072194a64772425966bf57a0116ebd17
* :yellow_heart: [Build #326085](https://buildkite.com/elastic/kibana-pull-request/builds/326085) was flaky 19c6bb4bf322ea6c5019f23664db93022abd9bcb
* :green_heart: [Build #325760](https://buildkite.com/elastic/kibana-pull-request/builds/325760) succeeded 7aab65151d2a410e9c3856021bee9f1adcf5636f

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327203","number":327203,"commit":"09db4a0ddf29dbe01b2572d6a8c18dc1c6849bcd"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326670","number":326670,"commit":"3de5c789072194a64772425966bf57a0116ebd17"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326085","number":326085,"commit":"19c6bb4bf322ea6c5019f23664db93022abd9bcb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/325760","number":325760,"commit":"7aab65151d2a410e9c3856021bee9f1adcf5636f"}],"number":327203}
buildkite-pr-comment-->

vitaliidm commented:
> Keep in mind that the point of this PR is not to "fix" any tests, but instead to identify, improve, and unskip a group of them that I believe will be made less flakey (or fixed entirely) by https://github.com/elastic/kibana/pull/229457.

Yes, I only was concerned about shards tests https://github.com/elastic/kibana/issues/224699, since they were failing constantly, unlike the rest of tests, which are flaky.

rylnd committed:
Merge branch 'main' into rylnd/unskip_with_better_failures

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/328396)
* Commit: d2d7843421b0834f17e800c3edf5ee80c7f97d2c

### Failed CI Steps
* [FTR Configs #27](https://buildkite.com/elastic/kibana-pull-request/builds/328396#01989998-e564-48c4-ba5a-3907f1062886)
* [FTR Configs #63](https://buildkite.com/elastic/kibana-pull-request/builds/328396#01989998-e594-4cf2-971d-b88ce3d20193)
* [FTR Configs #122](https://buildkite.com/elastic/kibana-pull-request/builds/328396#01989998-e5c4-407a-b245-f904c9451da9)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/328396#01989998-e564-48c4-ba5a-3907f1062886) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/328396/jobs/01989998-e564-48c4-ba5a-3907f1062886/artifacts/019899b8-a8fb-4a19-8444-67a73408f686) FTR Configs #<span></span>27 / InfraOps App feature controls infrastructure security global infrastructure read privileges infrastructure landing page with data shows Wafflemap
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/328396#01989998-e5c4-407a-b245-f904c9451da9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/328396/jobs/01989998-e5c4-407a-b245-f904c9451da9/artifacts/019899ce-525c-4e81-a46e-35ffbf81a36d) FTR Configs #<span></span>122 / Interactive setup APIs - Enrollment flow should be able to enroll with valid authentication code

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :yellow_heart: [Build #327203](https://buildkite.com/elastic/kibana-pull-request/builds/327203) was flaky 09db4a0ddf29dbe01b2572d6a8c18dc1c6849bcd
* :green_heart: [Build #326670](https://buildkite.com/elastic/kibana-pull-request/builds/326670) succeeded 3de5c789072194a64772425966bf57a0116ebd17
* :yellow_heart: [Build #326085](https://buildkite.com/elastic/kibana-pull-request/builds/326085) was flaky 19c6bb4bf322ea6c5019f23664db93022abd9bcb
* :green_heart: [Build #325760](https://buildkite.com/elastic/kibana-pull-request/builds/325760) succeeded 7aab65151d2a410e9c3856021bee9f1adcf5636f

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328396","number":328396,"commit":"d2d7843421b0834f17e800c3edf5ee80c7f97d2c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327203","number":327203,"commit":"09db4a0ddf29dbe01b2572d6a8c18dc1c6849bcd"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326670","number":326670,"commit":"3de5c789072194a64772425966bf57a0116ebd17"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326085","number":326085,"commit":"19c6bb4bf322ea6c5019f23664db93022abd9bcb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/325760","number":325760,"commit":"7aab65151d2a410e9c3856021bee9f1adcf5636f"}],"number":328396}
buildkite-pr-comment-->

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 230318` locally
cc: @rylnd
